integration-tests/src/test/java/com/datastax/oss/driver/core/metadata/DescribeIT.java

```diff
@@ -50,207 +51,112 @@
-    try {
-      InputStream is = DescribeIT.class.getResourceAsStream(resourceName);
-      if (is == null) {
-        // If no schema file is defined for tested cassandra version, just try 3.11.
-        if (CCM_RULE.getCassandraVersion().compareTo(Version.V3_0_0) >= 0) {
-          LOG.warn("Could not find schema file for {}, assuming C* 3.11.x", majorMinor);
-          is = DescribeIT.class.getResourceAsStream("/describe_it_test_3.11.cql");
-          if (is == null) {
-            throw new IOException();
-          }
-        }
-      }
-
-      closer.register(is);
-      ByteArrayOutputStream baos = new ByteArrayOutputStream();
-      PrintStream ps = new PrintStream(baos);
-      ByteStreams.copy(is, ps);
-      return baos.toString().replaceAll("ks_0", keyspace).trim();
-    } catch (IOException e) {
-      LOG.warn("Failure to read {}", resourceName, e);
-      fail("Unable to read " + resourceName + " is it defined?", e);
-    } finally {
-      try {
-        closer.close();
-      } catch (IOException e) { // no op
-        LOG.warn("Failure closing streams", e);
+  /**
+   * Find a creation script in our test resources that matches the current server version. If we
+   * don't have an exact match, use the closest version below it.
+   */
+  private File getScriptFile() {
+    URL logbackTestUrl = DescribeIT.class.getResource("/logback-test.xml");
+    if (logbackTestUrl == null || logbackTestUrl.getFile().isEmpty()) {
+      fail(
+          "Expected to use logback-test.xml to determine location of "
+              + "target/test-classes, but got URL %s",
+          logbackTestUrl);
+    }
+    File resourcesDir = new File(logbackTestUrl.getFile()).getParentFile();
+    File scriptsDir = new File(resourcesDir, isDse ? "DescribeIT/dse" : "DescribeIT/oss");
+    LOG.debug("Looking for a matching script in directory {}", scriptsDir);
+
+    File[] candidates = scriptsDir.listFiles();
+    assertThat(candidates).isNotNull();
+
+    File bestFile = null;
+    Version bestVersion = null;
+    for (File candidate : candidates) {
+      String fileName = candidate.getName();
+      String candidateVersionString = fileName.substring(0, fileName.lastIndexOf('.'));
+      Version candidateVersion = Version.parse(candidateVersionString);
+      LOG.debug("Considering {}, which resolves to version {}", fileName, candidateVersion);
+      if (candidateVersion.compareTo(serverVersion) > 0) {
+        LOG.debug("too high, discarding");
+      } else if (bestVersion != null && bestVersion.compareTo(candidateVersion) >= 0) {
+        LOG.debug("not higher than {}, discarding", bestVersion);
+      } else {
+        LOG.debug("best so far");
+        bestVersion = candidateVersion;
+        bestFile = candidate;
       }
     }
-    return "";
+    if (bestFile == null) {
+      fail("Could not find create script with version <= %s in %s", serverVersion, scriptsDir);
+      throw new AssertionError(); // never reached
```

Comment:
nit: We don't need this, do we? The test will fail on the line above if we get here.

Reply:
It's only here to avoid a compile-time error, the compiler doesn't know that `fail()` always throws.

But as I'm writing this I realize that there is a clearer way to do it: `assertThat(bestFile).isNotNull()`. I'll change it.

Reply:
That works too :+1: